### DISCUSSION TITLE
Conditionnal color on a table column

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#6341] [Help] [answered]
I want to put a special font color on the first column "status" of my table if the field value is "new".
How can I bind the field value to "new" ?

I tried for example {{ mytable.status }} or {{mytable.rows.status}} without success ...

**Comment by melohagan** [comment]
Hi @NicolasFabre81 

The only binding currently available within custom columns is the **Value** of that column for the particular row.
![Screenshot 2022-06-15 at 17 23 39](https://user-images.githubusercontent.com/101575380/173877496-cc41b2c9-9bf9-4775-a1fd-528f8f9e7b13.png)

There is no way to access all the row data presently within a custom column. 
It may be worth making the suggestion in this general table improvements issue: https://github.com/Budibase/budibase/issues/3273

**Reply by NicolasFabre81** [reply]
OK but how do I select the column that has to be considered for applying the condition ?
See what I did :
<img width="511" alt="2022-06-16 09_40_44-Budibase " src="https://user-images.githubusercontent.com/28592069/174018720-afdfeda9-a1f8-41a6-90c1-bbec51ac679e.png">

**Reply by melohagan** [reply]
Thanks for that! Configure columns will apply formatting to all values in that column. 

So there is currently no way to configure a cell. I've put that suggestion in this thread: https://github.com/Budibase/budibase/issues/3273

**Reply by NicolasFabre81** [reply]
Seems to be the same for the paragraph element : I'm not able to make a condition to align a repeated paragraph left or right depending on the repeater row index. Either all is left or all is right...

**Reply by brocococonut** [reply]
Conditional backgrounds can be done through a hack at the moment if you want. You can add an embed to the table which will add it to every row. In that embed you can have something like:
```javascript
const done = !!$("Sales Overview Table.Sales.Preapproval Done")

return `<style>
:root {
  --preapp-background: ${done ? 'inherit' : 'rgba(250, 25, 25, 0.5)'};
  --preapp-color: ${done ? 'inherit' : 'white'}
}
</style>`
```

Then you can go in to the column editor for the table and use the css variable like so:
![image](https://user-images.githubusercontent.com/6368283/174742045-68ed8aa2-fbca-4c57-9e02-4d20e755cfc6.png)
> `var(--preapp-background)`

It's not the most elegant solution, but it's the best that can be done at the moment given how restrictive tables can be. One drawback is you'll make sure to set the variable as a completely custom one that won't get used elsewhere on the page as the variable will still be set outside the table confines.

**Reply by melohagan** [reply]
@brocococonut Thank you very much for sharing :)

**Reply by rafikApp** [reply]
Hey @brocococonut @melohagan 
I used this javascript code to have a color which depends on the status, but it always returns the same color despite the difference of the status 

```
const done = "";

if ( $("CONSIGNATIONS Table.CONSIGNATIONS.statut") == "A confirmée" ){
	return `<style>
	
:root {
  --preapp-background: ${done ? 'inherit' : 'rgba(0, 0, 255, 0.5)'};
  --preapp-color: ${done ? 'inherit' : 'whith'}
}
</style>`
}
else {
	return `<style>
:root {
  --preapp-background: ${done ? 'inherit' : 'rgba(255, 0, 0, 0.5)'};
  --preapp-color: ${done ? 'inherit' : 'whith'}
}
</style>`
}
```

![Capture](https://user-images.githubusercontent.com/106697208/211940634-cf7eba63-8b42-4179-9ad4-1ce622d15a32.PNG)

**Reply by rafikApp** [reply]
I also tried to create a column (color) of type formula which will have a color which depends on the status
with the following code : 
```
const done = "";
return `<style>
:root {
  --preapp-background: ${done ? 'inherit' : $("CONSIGNATIONS Table.CONSIGNATIONS.color") };
  --preapp-color: ${done ? 'inherit' : 'whith'}
}
</style>`
```

but the same problem.

**Reply by melohagan** [reply]
Typo in 'white' you have 'whith'

**Reply by rafikApp** [reply]
Hey @melohagan thank you for your help.
I have the same problem.

```
const done = "";
switch($("CONSIGNATIONS Table.CONSIGNATIONS.statut")){
	case 'A confirmée':
		return `<style>
					:root {
						--preapp-background: ${done ? 'inherit' : 'rgba(0, 0, 255, 0.5)'};
						--preapp-color: ${done ? 'inherit' : 'white'}
					}
					</style>`
					break;
	case 'Consigné':
		return `<style>
					:root {
						--preapp-background: ${done ? 'inherit' : 'rgba(255, 0, 0, 0.5)'};
						--preapp-color: ${done ? 'inherit' : 'white'}
					}
					</style>`
					break;
	default : 
		return `<style>
					:root {
						--preapp-background: ${done ? 'inherit' : 'rgba(0, 255, 0, 0.5)'};
						--preapp-color: ${done ? 'inherit' : 'white'}
					}
					</style>`
					
}
```


![Capture1](https://user-images.githubusercontent.com/106697208/212024313-1e73f839-59f6-4220-850d-18df656d24dd.PNG)

**Reply by melohagan** [reply]
I'm not quite sure what could be wrong because I didn't write the solution, I have however a PR out for conditional table column color here: https://github.com/Budibase/budibase/pull/8512

It may not get merged into the main codebase, but you're very welcome to fork my `feature/conditional-table-cell` branch.

**Reply by rafikApp** [reply]
Thank you very much @melohagan 
But I didn't get how I can apply your method on my application

**Comment by xnohat** [comment]
This my hack for apply custom color format in table cell

Create an embed block in your screen with this code below
```
tableUnescapeHTMLFormat('table-component-name')

function tableUnescapeHTMLFormat(table_name){
	var table_elm = parent.document.querySelector(`[data-name=${table_name}`)

	if(!!table_elm){
		var table_heads = table_elm.firstChild.firstChild.firstChild.querySelector('div.spectrum-Table-head').querySelectorAll(':scope > *')
		var table_rows = table_elm.firstChild.firstChild.querySelector('div').querySelectorAll(':scope > *')

		//Scan every rows
		table_rows.forEach((row,idx)=>{
					//scan every cells of row and unescape html entities
					row.querySelectorAll(':scope > *').forEach((cell,idx)=>{
							//unescape html entities in all cell of table
							//https://stackoverflow.com/a/65592593
							var dummyTextArea = document.createElement('textarea');
							dummyTextArea.innerHTML = cell.firstChild.firstChild.innerHTML;
							cell.firstChild.firstChild.innerHTML = dummyTextArea.value;
					})
			})	
	}
}
```

Format column in Configure Column with plain HTML + CSS

![image](https://user-images.githubusercontent.com/4613808/223017390-aa803d21-d009-43bf-880f-a4f5bbc7488f.png)

![image](https://user-images.githubusercontent.com/4613808/223017458-8997b6dc-4527-49e1-a73f-10d41f1a9f91.png)

**Reply by NicolasFabre81** [reply]
Hello @xnohat , for me it does not work, it displays the HTML code inside the table :
<img width="688" alt="image" src="https://user-images.githubusercontent.com/28592069/227544499-6211a7c2-dc8f-4874-996a-f6a42e7f1ece.png">

**Reply by xnohat** [reply]
You must rename your table component to table-component-name or anything else. Then change this line

tableUnescapeHTMLFormat('table-component-name')

to your table-name

**Comment by melohagan** [comment, marked as answer]
Hey @NicolasFabre81 

You can use the Super Table to do this - I've added a short tutorial here: https://github.com/Budibase/budibase/discussions/10204
